craft_parts/sequencer.py

```diff
@@ -331,6 +345,44 @@ def _ensure_overlay_consistency(
         # execution should never reach this line
         raise RuntimeError(f"part {top_part!r} not in parts list")
 
+    def _check_if_dirty_on_overlay(self, part: Part, step: Step) -> bool:
+        """Verify whether the step is dirty because the overlay changed."""
+        if step == Step.OVERLAY:
+            # Layers depend on the integrity of its validation hash
+            current_layer_hash = self._layer_state.compute_layer_hash(part)
+            state_layer_hash = self._layer_state.get_layer_hash(part)
+            if current_layer_hash != state_layer_hash:
+                logger.debug("%s:%s changed layer hash", part.name, step)
```

Comment:
Here and in the other two `logger.debug` below: does it make sense to replace `step` in it? I mean, you already know which step it is, wouldn't it be simpler to do `logger.debug("%s:OVERLAY changed layer hash", part.name)`?

Also, I searched this PR to see how this line would look like IRL, but it seems it's missing from the tests?

Reply:
Added unit tests for this method. I kept the debug message so we can always have consistent step names even if we change the enumeration item name (impact should be negligible).